caikit_computer_vision/modules/object_detection/transformers.py

```diff
@@ -0,0 +1,191 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for transformers object detector transformer models.
+"""
+# Standard
+from typing import Union
+import os
+
+# Third Party
+from transformers import AutoImageProcessor, AutoModelForObjectDetection
+
+# First Party
+from caikit.core.exceptions import error_handler
+from caikit.core.modules import (
+    ModuleBase,
+    ModuleConfig,
+    ModuleLoader,
+    ModuleSaver,
+    module,
+)
+from caikit.interfaces.vision import data_model as caikit_dm
+from caikit.interfaces.vision.data_model.backends import image_pil_backend
+import alog
+
+# Local
+from ...data_model import BoundingBox, DetectedObject, ObjectDetectionResult
+from ...data_model.tasks import ObjectDetectionTask
+
+log = alog.use_channel("TRANSFORMERS_DETECT")
+error = error_handler.get(log)
+
+
+@module(
+    id="28dc918a-4e18-41c0-22a3-aa9c3c5caa11",
+    name="Transformers AutoModelForObjectDetection",
+    version="0.1.0",
+    task=ObjectDetectionTask,
+)
+class TransformersObjectDetector(ModuleBase):
+    _IMPROC_ARTIFACTS_CONFIG_KEY = "image_processor_artifacts"
+    _DETECTOR_ARTIFACTS_CONFIG_KEY = "object_detector_artifacts"
+
+    def __init__(
+        self,
+        model_name: str,
+        image_processor: "BaseImageProcessor",
+        detector_model: "PreTrainedModel",
+    ) -> "TransformersObjectDetector":
+        """Initialize a caikit-wrapped object detector."""
+        super().__init__()
+        self.model_name = model_name
+        self.image_processor = image_processor
+        self.detector_model = detector_model
+
+    @classmethod
+    def load(cls, model_path):
+        model_path = os.path.abspath(model_path)
```

Comment:
recently we have been started to move away from considering model_path strictly as `str` object, having this `abspath` might create problems in future. Can we instead just directly use `Moduleconfig.load` function?

Reply:
Done 😄